synapse/handlers/room.py

```diff
@@ -271,6 +272,15 @@ def create(etype, content, **kwargs):
 
             returned_events.append(power_levels_event)
 
+        if room_alias:
+            if (EventTypes.CanonicalAlias, '') not in initial_state:
```

Comment:
Probably don't need two levels of ``if`` block here:

    if room_alias and (EventTypes.CanonicalAlias, '') not in initial_state: